src/helpers.rs

```diff
@@ -199,56 +276,57 @@ fn encode_full_mappings(mappings: &[Mapping]) -> String {
-    if current_line < mapping.generated_line {
-      (0..mapping.generated_line - current_line).for_each(|_| out.push(';'));
-      current_line = mapping.generated_line;
-      current_column = 0;
-      initial = false;
-    } else if initial {
-      initial = false;
-    } else {
-      out.push(',');
-    }
-
-    encode(&mut out, mapping.generated_column, current_column);
-    current_column = mapping.generated_column;
-    if let Some(original) = &mapping.original {
-      active_mapping = true;
-      if original.source_index == current_source_index {
-        out.push('A');
       } else {
-        encode(&mut out, original.source_index, current_source_index);
-        current_source_index = original.source_index;
+        // No mapping is active
+        if mapping.original.is_none() {
+          // avoid writing unnecessary generated mappings
+          return acc;
+        }
       }
-      encode(&mut out, original.original_line, current_original_line);
-      current_original_line = original.original_line;
-      if original.original_column == current_original_column {
-        out.push('A');
+
+      out.clear();
+      if current_line < mapping.generated_line {
+        (0..mapping.generated_line - current_line).for_each(|_| out.push(';'));
+        current_line = mapping.generated_line;
+        current_column = 0;
+        initial = false;
+      } else if initial {
+        initial = false;
       } else {
-        encode(&mut out, original.original_column, current_original_column);
-        current_original_column = original.original_column;
+        out.push(',');
       }
-      if let Some(name_index) = original.name_index {
-        encode(&mut out, name_index, current_name_index);
-        current_name_index = name_index;
-        active_name = true;
+
+      encode(&mut out, mapping.generated_column, current_column);
+      current_column = mapping.generated_column;
+      if let Some(original) = &mapping.original {
+        active_mapping = true;
+        if original.source_index == current_source_index {
+          out.push('A');
+        } else {
+          encode(&mut out, original.source_index, current_source_index);
+          current_source_index = original.source_index;
+        }
+        encode(&mut out, original.original_line, current_original_line);
+        current_original_line = original.original_line;
+        if original.original_column == current_original_column {
+          out.push('A');
+        } else {
+          encode(&mut out, original.original_column, current_original_column);
+          current_original_column = original.original_column;
+        }
+        if let Some(name_index) = original.name_index {
+          encode(&mut out, name_index, current_name_index);
+          current_name_index = name_index;
+          active_name = true;
+        } else {
+          active_name = false;
+        }
       } else {
-        active_name = false;
+        active_mapping = false;
```

Comment:
The diff here is a little wired, I don't rename `active_name` to `active_mapping`.